data/adapters/eqtl_catalog_adapter.py

```diff
@@ -0,0 +1,157 @@
+
+
+class EQTLCatalog:
+    METADATA_PATH = 'data_loading_support_files/eqtl_catalog/tabix_ftp_paths.tsv'
+    ALLOWED_LABELS = ['qtl', 'study']
+
+    def __init__(self, filepath=None, label='qtl', writer: Optional[Writer] = None, **kwargs):
+        if label not in EQTLCatalog.ALLOWED_LABELS:
+            raise ValueError('Invalid label. Allowed values: ' +
+                             ','.join(EQTLCatalog.ALLOWED_LABELS))
+        self.filepath = filepath
+        self.label = label
+        self.type = 'edge'
+        self.writer = writer
+        self.source = 'eQTL Catalogue'
+        self.gene_validator = GeneValidator()
+
+    def process_file(self):
+        if self.label == 'qtl':
+            self.process_qtl()
+        elif self.label == 'study':
+            self.process_study()
+
+    def process_qtl(self):
+        dataset_id = self.filepath.split('/')[-1].split('.')[0]
+        found_dataset = False
+        with open(self.METADATA_PATH, 'r') as f:
+            metadata_reader = csv.reader(f, delimiter='\t')
+            next(metadata_reader)
+            for row in metadata_reader:
+                if row[1] == dataset_id:
+                    if row[8] == 'ge':
+                        label = 'eQTL'
+                        name = 'modulates expression of'
+                        inverse_name = 'expression modulated by'
+                    else:
+                        label = 'splice_QTL'
+                        name = 'modulates splicing of'
+                        inverse_name = 'splicing modulated by'
+                    biological_context = f'ontology_terms/{row[4]}'
+                    studay = f'studies/{row[0]}'
+                    simple_sample_summaries = [row[5]]
+                    # example: ftp://ftp.ebi.ac.uk/pub/databases/spot/eQTL/susie/QTS000001/QTD000001/QTD000001.credible_sets.tsv.gz
+                    source_url = row[10]
+                    found_dataset = True
+                    break
+        if not found_dataset:
+            raise ValueError(f'No metadata found for dataset {dataset_id}')
+
+        with gzip.open(self.filepath, 'rt') as f:
+            self.writer.open()
+            qtl_reader = csv.reader(f, delimiter='\t')
+            next(qtl_reader)
+            for row in qtl_reader:
+                variant_vcf_format = row[3]
+                chr, pos, ref_seq, alt_seq = variant_vcf_format.split('_')
+                variant_id = build_variant_id(
+                    chr, pos, ref_seq, alt_seq, 'GRCh38'
+                )
+                gene_id = row[1]
+                is_valid_gene_id = self.gene_validator.validate(
+                    gene_id)
+                if not is_valid_gene_id:
+                    continue
+                # this edge id is too long, needs to be hashed
+                variants_genes_id = hashlib.sha256(
+                    (variant_id + '_' + gene_id + '_' + biological_context).encode()).hexdigest()
```

Comment:
would use dataset_id instead of biological_context in key generation, seems the same tissue can be tested in different studies/datasets e.g. UBERON_0001013